pkg/controller/sparkapplication/controller.go

```diff
@@ -340,6 +338,15 @@ func (c *Controller) createSubmission(app *v1alpha1.SparkApplication) error {
 		appStatus.DriverInfo.WebUIPort = port
 	}
 
+	if app.Spec.Image == nil && c.operatorConfig.UnifiedSparkImage == "" {
```

Comment:
This check and the if block below should be moved into `onAdd`, and ideally we can add a method named `ValidateSparkApplication(app *SparkApplication, defaultSparkImage string)` in the same package as `types.go`, and we can name the file `validation.go`.